### PR TITLE
Add DoctrineMongoDBBundle recipe

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

File: doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+doctrine_mongodb:
+    auto_generate_proxy_classes: "%kernel.debug%"
+    auto_generate_hydrator_classes: "%kernel.debug%"
```
Comment (stof) [inline doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml]: I suggest using single quotes. Double quotes in YAML require escaping lots of things.

Comment (Lctrs) [inline doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml]: In the officiel recipe for `doctrine/doctrine-bundle`, we use double quotes for this parameter.

See https://github.com/symfony/recipes/blob/master/doctrine/doctrine-bundle/1.6/etc/packages/doctrine.yaml#L5.

File: doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    default_database: "%env(MONGODB_DBNAME)%"
+    document_managers:
+        default:
+            auto_mapping: true
+            mappings:
+                App:
+                    is_bundle: false
+                    type: annotation
+                    dir: "%kernel.project_dir%/src/Document"
+                    prefix: "App\\Document\\"
```
Comment (Pierstoval) [inline doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml]: We often voluntarily remove quotes for class names and would just use `prefix: App\Document\` in this case, it should work

File: doctrine/mongodb-odm-bundle/3.3/manifest.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+    "bundles": {
+        "Doctrine\\Bundle\\MongoDBBundle\\DoctrineMongoDBBundle": ["all"]
+    },
+    "copy-from-recipe": {
+        "etc/": "%ETC_DIR%/",
+        "src/": "%SRC_DIR%/"
+    },
+    "env": {
+        "MONGODB_URL": "mongodb://localhost:27017",
```
Comment (Pierstoval) [inline doctrine/mongodb-odm-bundle/3.3/manifest.json]: Can you suffix the database name to keep consistency with Doctrine ORM recipe and then simply have
`"MONGODB_URL": "mongodb://localhost:27017/symfony",` ?

**Comment by Lctrs** [comment]
One thing I don't know how to translate is the 2 lines that we needed to add to the old `app/autoload.php` (see the [docs](https://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html#register-the-annotations-and-the-bundle) for the lines in question). Any idea ?

File: doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml (hunk 2 above)
Comment (Lctrs) [inline doctrine/mongodb-odm-bundle/3.3/etc/packages/doctrine_mongodb.yaml]: That's not inline with the official Doctrine ORM recipe.

**Comment by Pierstoval** [comment]
The ORM annotation driver was lazy-loaded in symfony/symfony#21837, maybe something similar could be done for the ODM, I don't know 😕

**Comment by stof** [comment]
Actually, as the ODM bundle uses the annotation_reader defined by Symfony, it automatically benefits from the change done in Symfony.

**Comment by Lctrs** [comment]
Yeah. I tested it in a fresh application and it works as expected.

**Comment by Lctrs** [comment]
Should be good now.

**Comment by fabpot** [comment]
Thank you @Lctrs.

**Pull request closed without merging**